internal/bminventory/inventory.go

```diff
@@ -438,12 +453,30 @@ func (b *bareMetalInventory) GenerateClusterISO(ctx context.Context, params inst
 	jobName := fmt.Sprintf("createimage-%s-%s", cluster.ID, now.Format("20060102150405"))
 	imgName := getImageName(params.ClusterID)
 	log.Infof("Creating job %s", jobName)
-	if err := b.job.Create(ctx, b.createImageJob(jobName, imgName, ignitionConfig)); err != nil {
-		log.WithError(err).Error("failed to create image job")
-		return installer.NewGenerateClusterISOInternalServerError().
-			WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+	if b.Config.Target == "disconnected" {
+		cmd := exec.Command("python", "./data/install_process.py")
+		cmd.Env = append(os.Environ(),
+			"S3_ENDPOINT_URL="+b.S3EndpointURL,
+			"IGNITION_CONFIG="+ignitionConfig,
+			"IMAGE_NAME="+jobName,
+			"S3_BUCKET="+b.S3Bucket,
+			"aws_access_key_id="+b.AwsAccessKeyID,
+			"aws_secret_access_key="+b.AwsSecretAccessKey,
+			"WORK_DIR=/data",
+		)
+		var out bytes.Buffer
+		cmd.Stdout = &out
+		if err := cmd.Run(); err != nil {
+			log.Fatal(err)
+		}
+		log.Println(cmd.Stdout)
+	} else {
+		if err := b.job.Create(ctx, b.createImageJob(jobName, imgName, ignitionConfig)); err != nil {
+			log.WithError(err).Error("failed to create image job")
+			return installer.NewGenerateClusterISOInternalServerError().
+				WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+		}
 	}
-
 	if err := b.job.Monitor(ctx, jobName, b.Namespace); err != nil {
```

Comment:
I think you'll want to wrap the b.job.Monitor call inside the previous else clause.

Reply:
Yes, done.